src/pydocstyle/parser.py

```diff
@@ -250,311 +285,278 @@ def __init__(self, *args):
-        self.log.debug("finished parsing %s '%s'. Next token is %r (%s)",
-                       class_.__name__, name, self.current.kind,
-                       self.current.value)
-        return definition
-
-    def parse_skip_comment(self):
-        """Parse a definition comment for noqa skips."""
+            cls = Method if method else Function
+
+        docstring = self.get_docstring(node)
+        children = self.get_children(node, nested=True)
+
+        start = node.start_pos[0]
+        end = node.end_pos[0]
+        if start != end:
+            end -= 1
+
+        function = cls(name=str(node.name),
+                       start=start,
+                       end=end,
+                       decorators=self.get_decorators(node),
+                       docstring=docstring,
+                       children=children,
+                       parent=None,
+                       skipped_error_codes=self.parse_skip_comment(node))
+
+        for child in function.children:
+            child.parent = function
+
+        return [function]
+
+    def handle_class(self, node, nested=False, *args, **kwargs):
+        cls = NestedClass if nested else Class
+
+        docstring = self.get_docstring(node)
+        children = self.get_children(node, nested=True, method=True)
+
+        start = node.start_pos[0]
+        end = node.end_pos[0]
+        if start != end:
+            end -= 1
+
+        klass = cls(name=str(node.name),
+                    start=start,
+                    end=end,
+                    decorators=self.get_decorators(node),
+                    docstring=docstring,
+                    children=children,
+                    parent=None,
+                    skipped_error_codes=self.parse_skip_comment(node))
+
+        for child in klass.children:
+            child.parent = klass
+
+        return [klass]
+
+    def parse_skip_comment(self, node):
+        """Parse a "# noqa" comment for definitions."""
+        for child in node.children:
+            if child.type == 'operator' and child.value == ':':
+                break
+        else:
+            return ''
+        line_number = child.start_pos[0]
+        line = self.source[line_number - 1]
+        line_stream = StringIO(line)
+        # The line always ends with an NEWLINE and ENDMARKER, so we take
+        # the third one from the end and check if it's a comment.
+        try:
+            token = list(tk.generate_tokens(line_stream.readline))[-3]
+        except tk.TokenError:
+            return ''
+        if token[0] != tk.COMMENT:
+            return ''
+        comment = token[1]
         skipped_error_codes = ''
-        if self.current.kind == tk.COMMENT:
-            if 'noqa: ' in self.current.value:
-                skipped_error_codes = ''.join(
-                     self.current.value.split('noqa: ')[1:])
-            elif self.current.value.startswith('# noqa'):
-                skipped_error_codes = 'all'
+        if 'noqa: ' in comment:
```

Comment:
If I stripped the `noqa` logic into a separate library, would you use that instead?